modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -258,10 +263,14 @@ public boolean containsAll(Collection<?> c) {
     /** {@inheritDoc} */
     @Override
     public boolean addAll(Collection<? extends T> c) {
+        if (this.getClass() != c.getClass()) {
```

Comment:
same, handling of the null 'c' argument.
I wonder if this is intended behavior?

Reply:
I doubt the `Set` contract was breached here on purpose. More likely, the original implementation was never exposed, or the original developer didn't realize that `Set` has a very specific contract that you must follow, or it won't interact well with other collections.

Reply:
To be clear, `BitSet` is currently exposed via `Node#getPseudoClassStates`.

I suspect it is almost never used by anyone, because there are a lot of bugs.

For example, equality won't work correctly:

     node.getPseudoClassStates().equals(Set.of( ... ));  // Always false, sets are of different types

Nor will `containsAll`:

     node.getPseudoClassStates().containsAll(Set.of( ... ));  // Always false, because Set is not a BitSet
     node.getPseudoClassStates().containsAll(null);  // Doesn't throw NPE, returns false

All of these are `Set` contract violations, that I suppose were thought to be irrelevant because it is a com.sun.* class.